pkg/util/logs/logs.go

```diff
@@ -4,41 +4,31 @@
 package logs
 
 import (
-	"flag"
-	"log"
-	"time"
-
-	"github.com/golang/glog"
-	"github.com/spf13/pflag"
-	"k8s.io/apimachinery/pkg/util/wait"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	kzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
-var logFlushFreq = pflag.Duration("log-flush-frequency", 5*time.Second, "Maximum number of seconds between log flushes")
-
-// GlogWriter serves as a bridge between the standard log package and the glog package.
-type GlogWriter struct{}
-
-// Write implements the io.Writer interface.
-func (writer GlogWriter) Write(data []byte) (n int, err error) {
-	glog.Info(string(data))
-	return len(data), nil
-}
-
-// InitLogs initializes logs the way we want for kubernetes.
-func InitLogs() {
-	log.SetOutput(GlogWriter{})
-	log.SetFlags(0)
-	flag.Set("logtostderr", "true")
-	// The default glog flush interval is 30 seconds, which is frighteningly long.
-	go wait.Until(glog.Flush, *logFlushFreq, wait.NeverStop)
-}
-
-// FlushLogs flushes logs immediately.
-func FlushLogs() {
-	glog.Flush()
-}
-
-// NewLogger creates a new log.Logger which sends logs to glog.Info.
-func NewLogger(prefix string) *log.Logger {
-	return log.New(GlogWriter{}, prefix, 0)
+// InitLogs initializes logs with Time and Global Level of Logs set at Info
+func InitLogs(opts kzap.Options) {
+	var config zap.Config
+	if opts.Development {
+		config = zap.NewDevelopmentConfig()
+	} else {
+		config = zap.NewProductionConfig()
+	}
+	if opts.Level != nil {
+		config.Level = opts.Level.(zap.AtomicLevel)
+	} else {
+		config.Level.SetLevel(zapcore.InfoLevel)
+	}
+	config.EncoderConfig.EncodeTime = zapcore.RFC3339TimeEncoder
+	config.EncoderConfig.TimeKey = "@timestamp"
```

Comment:
is @timestamp going to show up in the log message

Reply:
yes, the original filebeat message is like
```
 {
    message: "{ message: 'hello', @timestamp: <timestamp of logging, this @timestamp>}" //json as string
    @timestamp: //timestamp of sending the file
 }
```
the filebeat processor will process/parse the json string to the following to send to elasticsearch:
```
 {
    log: {
      message: 'hello',
    }
    @timestamp: //timestamp of logging, this @timestamp
 }
```

Reply:
so what does a log message look like now in the verrazzano-operator log file

Reply:
```
{"level":"info","@timestamp":"2020-11-05T15:38:22-05:00","caller":"logs/logs_test.go:19","message":"greeting hello"}
```